test/sqllogictest/timezone.slt

```diff
@@ -23,3 +23,64 @@ SET TimeZone = 'uTc'
 
 statement error parameter TimeZone can only be set to UTC
 SET TIME ZONE bad
+
+query T
+SELECT TIMESTAMP '2020-12-21 18:53:49' AT TIME ZONE TEXT 'America/New_York'
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT TIMESTAMPTZ '2020-12-21 18:53:49 America/New_York'
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT timezone(TEXT 'America/New_York', TIMESTAMP '2020-12-21 18:53:49')
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT TIMESTAMP '2020-12-21 18:53:49' AT TIME ZONE TEXT 'America/New_York'
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT TIMESTAMPTZ '2020-12-21 18:53:49 America/New_York'
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT timezone(TEXT 'America/New_York', TIMESTAMP '2020-12-21 18:53:49')
+----
+2020-12-21 23:53:49+00
+
+query T
+SELECT timezone(INTERVAL '+11'HOUR, TIME '18:53:49')::time
+----
+05:53:49
+
+query T
+SELECT TIMESTAMPTZ '2020-12-21 18:53:49 Pacific/Auckland' AT TIME ZONE TEXT 'Turkey'
+----
+2020-12-21 08:53:49
+
+query T
+SELECT TIMESTAMPTZ '2007-02-01 00:00:00+5:30:16';
+----
+2007-01-31 18:29:44+00
+
+statement error timezone interval must not contain months or years
+SELECT timezone(INTERVAL '+11'MONTH, TIME '18:53:49')
+
+statement error invalid input syntax for timestamptz: Invalid timezone string \(\+16:60\): timezone hour out of range 16
+SELECT TIMESTAMPTZ '2020-01-01 00:00:00+16:60'
+
+# TODO: Postgres returns 2020-03-08 07:00:00+00 for both 02:00:00 and 03:00:00, chrono-tz refuses.
```

Comment:
PSQL's and chrono-tz's timezone/offset resolution methods disagree when it comes to switching clocks.

Reply:
hmm yeah that's a bit odd, I'd expect if anything that the other time (3:00) would be the one that is allowed. Latest might allow that, indeed.

It doesn't make sense for the same second to occur twice, clocks go directly from 1:59:59 -> 3:00:00. For now I think it would be fine to just open an issue saying that we should accept both ambiguous times at DST boundaries.